### PR TITLE
add new hub component download

### PR DESCRIPTION
**Pull request by lautarolecumberry**

### CODE REVIEW

**Review by matisilva — APPROVED** [review]

File: splight_lib/client/hub/abstract.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def _get(
         pass
 
     @abstractmethod
-    def upload(self, data: Dict, files: Dict) -> Tuple:
+    def upload(self, id: str, file_path: str, type: str) -> BaseModel:
```
Comment (marianogaragiola) [inline splight_lib/client/hub/abstract.py:24]:
```suggestion
    def upload(self, id: str, file_path: str, type: str):
```
Change to follow the implementation. Also, in the clients of the library we are not using the pydantic BaseModel, only python built-in types

File: splight_lib/client/hub/client.py [hunk 2]
```diff
@@ -63,27 +66,58 @@ def _get(
     def get_org_id(self):
         return self._org_id
 
-    def upload(self, data: Dict, files: Dict) -> Tuple:
-        url = self._hub_url / "upload/"
+    def _create(self, instance: BaseModel) -> BaseModel:
```
Comment (marianogaragiola) [inline splight_lib/client/hub/client.py:69]: Do not use pydantic BaseModel, use the model converted to dict

File: splight_lib/models/hub.py [hunk 3]
```diff
@@ -191,15 +212,18 @@ def upload(cls, path: str):
...
+                id=hub_component.id, file_path=readme_path, type="readme"
+            )
         except Exception as exc:
             raise exc
         finally:
             if os.path.exists(file_name):
                 os.remove(file_name)
-        return cls.model_validate(component)
+
+        hub_component.build()
```
Comment (marianogaragiola) [inline splight_lib/models/hub.py:228]: This could raise an exception also, it should be within the try block

File: splight_lib/models/hub.py [hunk 4]
```diff
@@ -147,15 +147,36 @@ def delete(self):
...
+        )
+        if not raw_hub_component:
+            raw_hub_component = HubComponent(
+                name=name,
+                version=version,
+                splight_lib_version=spec.get("splight_lib_version", None),
+                splight_cli_version=spec.get("splight_cli_version", None),
+                privacy_policy=spec.get("privacy_policy"),
+                component_type=spec.get("component_type"),
+            ).save()
```
Comment (marianogaragiola) [inline splight_lib/models/hub.py:177]: What is happening with routines, input, and other confirations of the hub component, are those needed?

File: splight_lib/models/hub.py [hunk 5]
```diff
@@ -147,15 +147,36 @@ def delete(self):
 
     def download(self):
         hub_client = get_hub_client()
-        params = {"name": self.name, "version": self.version}
-        return hub_client.download(data=params)
+        return hub_client.download(id=self.id, name=self.name, type="source")
+
+    def save(self):
+        hub_client = get_hub_client()
+        return hub_client.save(instance=self)
```
Comment (marianogaragiola) [inline splight_lib/models/hub.py:154]:
```suggestion
        return hub_client.save(instance=self.model_dump())
```
Check the client always get the model in the dict format. That is the approach used in the public library

File: splight_lib/client/hub/client.py [hunk 6]
```diff
@@ -93,4 +127,7 @@ def delete(self, id: str) -> None:
         ), f"Failed to delete component: {response.json()}"
 
     def save(self, instance: BaseModel) -> BaseModel:
```
Comment (marianogaragiola) [inline splight_lib/client/hub/client.py:129]: This is an error in the library, it should be
```python
def save(self, intance: Dict) -> Dict:
   ...
```

Comment (marianogaragiola) [reply splight_lib/client/hub/client.py:129]: Check the database client for reference

**Review by marianogaragiola — APPROVED** [review]

**Pull request merged**